rdr_service/services/duplicate_detection.py

```diff
@@ -0,0 +1,80 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import List
+
+from sqlalchemy.orm import Session
+
+from rdr_service.dao.base_dao import with_session
+from rdr_service.dao.duplicate_account_dao import DuplicateAccountDao, DuplicationSource, DuplicateExistsException
+from rdr_service.model.participant_summary import ParticipantSummary
+
+
+class _SummaryCache:
+    def __init__(self, session: Session):
+        self._session = session
+        self._name_dob_cache = defaultdict(         # top level dict for keying by first name
+            lambda: defaultdict(                    # next level dict for keying by last name
+                lambda: defaultdict(                # next level dict for keying by dob
+                    list                            # list to hold any participant ids that have the above data
+                )
+            )
+        )
+        self._email_cache = defaultdict(list)
+        self._phone_cache = defaultdict(list)
+        self._load_data()
+
+    def _load_data(self):
+        for summary in DuplicateAccountDao.query_participant_duplication_data(self._session):
```

Comment:
should we throw in a try, except here incase any exceptions occur when trying to load the data we can log, and handle accordingly?

Reply:
If any exceptions happen when loading this data we'd need to kill the whole run anyway. Since we wouldn't need to do any other cleanup, and since we'd re-raise the exception we catch here (so it can stop the rest of the process and so it can get logged by the error handler) it seems like it makes more sense to not catch it here.